x-pack/filebeat/input/awss3/s3.go

```diff
@@ -330,6 +335,14 @@ func (p *s3Poller) Purge() {
 		lock.(*sync.Mutex).Unlock()
 		p.workersListingMap.Delete(listingID)
 		p.states.DeleteListing(listingID)
+
+		// Listing is removed from all states, we can finalize now
+		for _, state := range states {
+			processor, _ := p.createS3ObjectProcessor(context.TODO(), *state)
```

Comment:
can we use the same `context` as [here](https://github.com/elastic/beats/pull/33559/files#diff-c6357a5d68a965123a3b3dc3cbf0bddfe7a5628dceb5327885da33f1735a3facR226)?

Feel free to pass it down to `s3Poller.Purge()` as argument in case